src/modules/mainFunction.js

```diff
@@ -64,26 +66,58 @@ class TaskList {
     this.tasksContainer.addEventListener('click', (e) => {
       const taskItem = e.target.closest('.taskItems');
       if (taskItem) {
-        const infoTask = taskItem.querySelector('.infoTask');
-        infoTask.addEventListener('blur', () => {
-          const newDescription = infoTask.value.trim();
+        const descriptionElement = taskItem.querySelector('.infoTask');
+        descriptionElement.addEventListener('blur', () => {
+          const newDescription = descriptionElement.value.trim();
           if (newDescription) {
-            const id = parseInt(infoTask.dataset.index, 10);
+            const id = parseInt(descriptionElement.dataset.index, 10);
             const taskIndex = this.tasksInfo.findIndex((task) => task.index === id);
             this.tasksInfo[taskIndex].description = newDescription;
           }
           this.display();
+          this.checkStatus();
           this.saveTasks();
         });
       }
     });
   }
 
+  checkStatus = () => {
+    const checkboxes = document.querySelectorAll('.checkbox');
+    checkboxes.forEach((checkbox) => {
+      checkbox.addEventListener('click', () => {
+        const id = parseInt(checkbox.closest('.taskItems').id.split('-')[1], 10);
+        const taskIndex = this.tasksInfo.findIndex((task) => task.index === id);
+        this.tasksInfo[taskIndex].completed = checkbox.checked;
+        this.saveTasks();
+        const infoTask = checkbox.closest('.taskItems').querySelector('.infoTask');
+        if (checkbox.checked) {
+          infoTask.style.textDecoration = 'line-through';
+        } else {
+          infoTask.style.textDecoration = 'none';
+        }
+      });
+    });
+  }
+
+  deleteCompletedTasks() {
```

Comment:
- [ ] According to the project requirements, The updated array of items must be stored in local storage, so the user gets the correct list values after the page reloads. However, all the checked books (completed: true) get deleted from the local storage all the time I reload the browser. Kindly prevent this 👍🏻